src/main/kotlin/com/exactpro/th2/codec/AbstractCodecProcessor.kt

```diff
@@ -135,7 +136,7 @@ abstract class AbstractCodecProcessor<BATCH, GROUP, MESSAGE>(
             }
         }
 
-        val parentEventIds: Sequence<EventID> = if (useParentEventId) messageGroup.eventIds else emptySequence()
+        val pairIds: Sequence<Pair<MessageID, EventID?>> = if (useParentEventId) messageGroup.pairIds(batch) else emptySequence()
```

Comment:
In almost all places where the `pairIds` is used we use `toMap` method. Maybe it would be better just to return `Map<MessageID, EventID?>` from `pairIds` method.

Reply:
I continue work with stream because the result of pairIds is optionally used

Reply:
So, instead of creating a map once and either throwing it away without using it in some cases or using this single map everywhere you would prefer to create the map multiple times in some cases and after each usage throw it away?

Okay, I have no objections.

There is also another way to initialize something only if it is needed
```kotlin
fun main() {
  val lazyHavyMap by lazy { havyMap() }
  println("Map is not yet created")
  println(lazyHavyMap)
}

fun havyMap(): Map<String, Any> {
    println("Creating map")
    return mapOf("test" to 42)
}
```

Reply:
Thank you for example